guides_and_docs/virt_env_Fram_with_VScode.md

```diff
@@ -0,0 +1,75 @@
+# How to use Virtual studio Code and Fram
+
+If you like to run significant bits of code, you probably want to do it on a Server. I decided to run my code on Fram, one of the Sigma2 HPCs. 
+
+I use Visual Studio Code (VS code) to connect to Fram and to run my scripts. To connect to any remote via ***ssh***, please follow the [instructions](https://code.visualstudio.com/docs/remote/ssh) on the VS code documentation. If you want to connect to Fram via VS code, use `<username>@fram.sigma2.no`, while Nird needs to know about the individual nodes. Nird: `<username>@loginX.nird.sigma2.no` where X is the node 0, 1, 2, or 3.
+
+> **_NOTE:_**  The problem is that login.nird.sigma2.no is an alias corresponding to 4 login nodes:
+> - login0.nird.sigma2.no
+> - login1.nird.sigma2.no
+> - login2.nird.sigma2.no
+> - login3.nird.sigma2.no
+>
+> and that probably won't work with VS code unless if you stick to one login node and use specifically that one.
+
+
+Once we are logged in to Fram, we will be in our home directory. Open the terminal with one of the below commands:
+- Use the `Ctrl+` ` keyboard shortcut with the backtick character.
+- Use the **View > Terminal** menu command.
+- From the **Command Palette** (`Ctrl+Shift+P`), use the **View: Toggle Integrated Terminal** command. 
+  
+To check if you are in your home directory, type the following:
+```
+$ cd ~
+$ pwd
+/cluster/home/username
+``` 
+Since I want my science to be reproducible, I create my personal virtual environment for Python on Fram. [Why do I want to use a private virtual environment?](https://towardsdatascience.comwhy-you-should-use-a-virtual-environment-for-every-python-project-c17dab3b0fd0)
```

Comment:
Since it is strongly recommended to make your science [...], you ought to create your [...]. I -> you